lib/autoproj/package_set.rb

```diff
@@ -4,6 +4,45 @@ module Autoproj
     # package definitions (.autobuild files), and finally definition of
     # dependencies that are provided by the operating system (.osdeps file).
     class PackageSet
+        # Comparator can be used to compare the list of package sets
+        # so that they can be sorted according to the following constraints:
+        # 1. dependencies among the package sets
+        # 2. order in which they are defined in the main configuration (manifest)
+        #
+        # Note that 1. naturally always takes precedence over 2. since 2.
+        # only affects a sorting of package sets that have no dependency
+        class Comparator
+            attr_reader :pkg_sets, :main_imports
```

Comment:
Would you consider renaming `pkg_sets` into something like `imports_of` ? The name has really confused me.